src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs

```diff
@@ -596,5 +600,160 @@ private static void GuardBlockIndex(int idx)
             DebugGuard.MustBeLessThan(idx, Size, nameof(idx));
             DebugGuard.MustBeGreaterThanOrEqualTo(idx, 0, nameof(idx));
         }
+
+        /// <summary>
+        /// Transpose the block into the destination block.
+        /// </summary>
+        /// <param name="d">The destination block</param>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public void TransposeInto(ref Block8x8F d)
+        {
+#if SUPPORTS_RUNTIME_INTRINSICS
+            if (Avx.IsSupported)
+            {
+                this.TransposeIntoAvx(ref d);
+            }
+            else
+#endif
+            {
+                this.TransposeIntoFallback(ref d);
+            }
+        }
+
+#if SUPPORTS_RUNTIME_INTRINSICS
+        /// <summary>
+        /// AVX-only variant for executing <see cref="TransposeInto(ref Block8x8F)"/>.
+        /// <see href="https://stackoverflow.com/questions/25622745/transpose-an-8x8-float-using-avx-avx2/25627536#25627536"/>
+        /// </summary>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public void TransposeIntoAvx(ref Block8x8F d)
+        {
+#if avxvariant1
+            Vector256<float> r0 = Unsafe.As<Vector4, Vector256<float>>(ref this.V0L);
+            Vector256<float> r1 = Unsafe.As<Vector4, Vector256<float>>(ref this.V1L);
+            Vector256<float> r2 = Unsafe.As<Vector4, Vector256<float>>(ref this.V2L);
+            Vector256<float> r3 = Unsafe.As<Vector4, Vector256<float>>(ref this.V3L);
+            Vector256<float> r4 = Unsafe.As<Vector4, Vector256<float>>(ref this.V4L);
+            Vector256<float> r5 = Unsafe.As<Vector4, Vector256<float>>(ref this.V5L);
+            Vector256<float> r6 = Unsafe.As<Vector4, Vector256<float>>(ref this.V6L);
+            Vector256<float> r7 = Unsafe.As<Vector4, Vector256<float>>(ref this.V7L);
+
+            Vector256<float> t0 = Avx.UnpackLow(r0, r1);
```

Comment:
I'd expect there are a few spills as part of this due to having 8-16 live registers at a time.

You might see better perf by intermixing the three steps given here. That is, handle
t0, t2, t4, t6 to produce r0, r1, r4, r5 and store them
and then handle t1, t3, t5, t7 to produce r2, r3, r6, r7 and store them

I imagine this will help minimize the number of stack spills that happen and may also allow better pipelining on the latest CPUs (without hindering older CPUs)

Reply:
Noting that I did not check the disassembly here, I am just speculating based on past experience 😄

Reply:
It helped! 🎉